### PR TITLE
Adds documentation about how to create releases in BCR.

### LINKED ISSUE #153: Explain how to release to BCR
## Desired behavior
Explain the process of releasing to BCR

## Alternatives considered
N/A

## Implementation suggestion
Use the same release page or create a new one with release notes for maintainers

## Additional context
N/A

### PR DESCRIPTION
**Pull request by agalbachicar**
# 🎉 New feature

Closes #153 

## Summary

Completes the documentation to create BCR releases.

## Test it
N/A

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: docs/releases.rst [hunk 1]
```diff
@@ -143,3 +143,61 @@ Create a new package patch release
...
+
+ROS Build Farm
+^^^^^^^^^^^^^^
+
+TODO(https://github.com/maliput/maliput_documentation/issues/156)
+
+Bazel Central Registry
+^^^^^^^^^^^^^^^^^^^^^^
+
+The Github maliput organization hosts a `fork <https://github.com/maliput/bazel-central-registry>` of `bazel-central-registry <https://github.com/bazelbuild/bazel-central-registry>`. Upon pushing the tag,
```
Comment (francocipollone) [inline docs/releases.rst:181]: We are not relying anymore on the bcr fork

**Review by francocipollone — CHANGES_REQUESTED** [review]
Thanks for pushing this

File: docs/releases.rst (hunk 1 above)
Comment (agalbachicar) [reply docs/releases.rst:181]: The PR in upstream is created from the BCR fork if I understood it correctly.

Look at https://github.com/bazelbuild/bazel-central-registry/pull/1907 and the base [branch](https://github.com/maliput/bazel-central-registry/tree/maliput/maliput%401.3.0-791c220d).

Am I missing something?

Comment (francocipollone) [reply docs/releases.rst:181]: Sorry, I wasn't clear.

At some point in time we were relying directly on our BCR fork. Nowadays the releases are targeting the original BCR repo. The fork now, only serves as our workspace for creating PRs against official BCR.
When first read this paragraph I felt you were mentioning that the had our own BCR (and we don't use official BCR)

File: docs/releases.rst [hunk 2]
```diff
@@ -143,3 +143,61 @@ Create a new package patch release
...
+and the release page creation, a PR with the bazel configuration files is created to register a new release in BCR.
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply docs/releases.rst:182]: Go for it

**Review by francocipollone — APPROVED** [review]

**Review by francocipollone — APPROVED** [review]

**Pull request merged**